nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Not Implmented'
+  end
+end
+
+class Decorator < Nameable
```

Comment:
- [ ] [optional] @tarikbouari kindly move every class to its own file, this will make your code neater and easy to scale in the future.